debug_assert.hpp

```diff
@@ -44,7 +44,18 @@
 #endif
 
 #ifndef DEBUG_ASSERT_ASSUME
-#ifdef __GNUC__
+#ifdef __clang__
+#if __has_builtin(__builtin_assume)
+#define DEBUG_ASSERT_ASSUME(Expr) __builtin_assume(Expr)
```

Comment:
Can you use `defined(__clang__) && __has_builtin...` there and then just have a single `#elif defined(__GNUC__)`? Avoids the repetition.